STYLEGUIDE.md

```diff
@@ -0,0 +1,133 @@
+
+# Typedefs styleguide
+
+This styleguide aims to answer questions relevants to the cosmetics related to 
+writing Idris programs.
+
+Since idris is a relatively young language, there is no general convention 
+about its style. This document should serve as reference for current and future 
+users of the typedefs codebase.
+
+## Line length: 120
+
+The maximum line length is 120 characters. In today's wide screen monitors 
+there is no reason to limit ourselves to 80 characters, yet, having a maximum
+limit helps avoiding degenerate situations.
+
+## Indent style: 2 spaces
+
+Idris makes uses of some whitespace sensitive construction (typically the `do`
+notation), and also inherits from a culture of ascii-art code from Agda and
+Haskell. In order to allow this level of visual customisation and ensure
+consistency across editors, spaces are used.
+
+The language does not benefit from more than 2 space of indentation in most
+cases. Thought special constructions lend themselve to more spaces for 
+indentation.
+
```

Comment:
Following the Halogen-inspired style we use elsewhere, I propose the following for type sigs:

---
### Indentation: type signatures

```idris
fn
   : A
  -> B
  -> c
fn x = 42
```

Reply:
Personally, I think this looks horrible, but I could live with it. (Some passive aggression there for you.) Even worse, the example doesn't typecheck!

Reply:
@fredrikNordvallForsberg Thank you for your insightful and introspective feedback! I hope you will find the following amendments soothing:
```idris
-- short type sig, so no need for nasty newlineses
whatever : Int -> Int
whatever x = 42

-- but then, look at all the pretty alignment here!
numberOfTheBeast
   : AnInordinatelyLongTypeInTheHopesOfAppeasingFredrik
  -> YetAnotherExcruciatinglyLongType_ThisBetterWork
  -> Int
numberOfTheBeast x y =
  666
```

Reply:
I'm currently writing multiline function signatures as

```
fn :
     a : Type
  -> b : Type
  -> Type
```

I guess that having a line with ` : a : Type` could be not that clear at first sight

Reply:
@marcosh I think your suggestion makes sense. Moreover the style stays consistent when you introduce type classes

```idris
fn : (myTypeClasss a)
  => a : Type
  -> b : Type
  -> Type
```

I'll add two paragraphs about function signatures and interface declarations. As well as argument grouping